sn_cli/src/subcommands/files/mod.rs

```diff
@@ -231,11 +227,15 @@ async fn upload_files(
     // keep track of the progress in a separate task
     let progress_bar_clone = progress_bar.clone();
     let total_existing_chunks_clone = total_existing_chunks.clone();
-    tokio::spawn(async move {
+
+    let progress_handler = tokio::spawn(async move {
+        let mut upload_terminated_with_error = false;
+        // The channel is closes if the upload completes/errors out, so the loop is guaranteed to end.
```

Comment:
// The loop is guaranteed to be ended, as the channel will be closed when the upload completed or errored out.